tests/tests/es7.promise.finally.ls

```diff
@@ -0,0 +1,29 @@
+{module, test} = QUnit
+module \ES7
+
+test 'Promise#finally' (assert)!->
+  assert.isFunction Promise::finally
+  NATIVE and assert.arity Promise::finally, 2 # FF - 0
+  assert.name Promise::finally, \_finally
+  assert.looksNative Promise::finally
+  assert.nonEnumerable Promise::, \finally
+  # subclassing, @@species pattern
+  promise = new Promise !-> it 42
+  promise@@ = FakePromise1 = !-> it ->, ->
+  FakePromise1[Symbol?species] = FakePromise2 = !-> it ->, ->
+  assert.ok promise.finally(->) instanceof FakePromise2, 'subclassing, @@species pattern'
+  # subclassing, incorrect `this.constructor` pattern
+  promise = new Promise !-> it 42
+  promise@@ = FakePromise1 = !-> it ->, ->
+  assert.ok promise.finally(->) instanceof Promise, 'subclassing, incorrect `this` pattern'
+  # NewPromiseCapability validations
+  promise = new Promise !-> it 42
+  promise@@ = FakePromise1 = !-> it ->, ->
+  FakePromise1[Symbol?species] = !->
+  assert.throws (!-> promise.finally(->)), 'NewPromiseCapability validations, #1'
+  FakePromise1[Symbol?species] = !-> it null, ->
+  assert.throws (!-> promise.finally(->)), 'NewPromiseCapability validations, #2'
+  FakePromise1[Symbol?species] = !-> it ->, null
+  assert.throws (!-> promise.finally(->)), 'NewPromiseCapability validations, #3'
```

Comment:
I see only subclassing tests. For `Promise#then` it's enough because of the main test case for them - https://github.com/promises-aplus/promises-tests, but `finally` missed here.